CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLHarvester.cc

```diff
@@ -0,0 +1,476 @@
+    double e0 = f1->GetParError(0);
+    double p1 = f1->GetParameter(1);
+    double e1 = f1->GetParError(1);
+    double p2 = f1->GetParameter(2);
+    double e2 = f1->GetParError(2);
+    double p3 = f1->GetParameter(3);
+    double e3 = f1->GetParError(3);
+    double p4 = f1->GetParameter(4);
+    double e4 = f1->GetParError(4);
+    double p5 = f1->GetParameter(5);
+    double e5 = f1->GetParError(5);
+    double chi2 = f1->GetChisquare();
+    double prob = f1->GetProb();
+
+    edm::LogPrint("LorentzAngle") << std::setprecision(4) << hists.BPixnewModule_[j] << "\t" << hists.BPixnewLayer_[j]
+                                  << "\t" << p0 << "\t" << e0 << "\t" << p1 << std::setprecision(3) << "\t" << e1
+                                  << "\t" << e1 / p1 * 100. << "\t" << (p1 - 0.424) / e1 << "\t" << p2 << "\t" << e2
+                                  << "\t" << p3 << "\t" << e3 << "\t" << p4 << "\t" << e4 << "\t" << p5 << "\t" << e5
+                                  << "\t" << chi2 << "\t" << prob << "\t" << hists.BPixnewDetIds_[j] << std::endl;
+  }
+
+  //loop over modlues and layers to fit the lorentz angle
+  for (int i_layer = 1; i_layer <= hists.nlay; i_layer++) {
+    for (int i_module = 1; i_module <= hists.nModules_[i_layer - 1]; i_module++) {
+      int i_index = i_module + (i_layer - 1) * hists.nModules_[i_layer - 1];
+      //loop over bins in depth (z-local-coordinate) (in order to fit slices)
+      for (int i = 1; i <= hist_depth_; i++) {
+        findMean(h_drift_depth_adc_slice_, i, i_index);
+      }  // end loop over bins in depth
+
+      f1->SetParameter(0, 0);
+      f1->SetParError(0, 0);
+      f1->SetParameter(1, 0.4);
+      f1->SetParError(1, 0);
+      f1->SetParameter(2, 0.0);
+      f1->SetParError(2, 0);
+      f1->SetParameter(3, 0.0);
+      f1->SetParError(3, 0);
+      f1->SetParameter(4, 0.0);
+      f1->SetParError(4, 0);
+      f1->SetParameter(5, 0.0);
+      f1->SetParError(5, 0);
+      f1->SetChisquare(0);
+
+      hists.h_mean_[i_index]->getTH1()->Fit(f1.get(), "ERQ");
+      double p0 = f1->GetParameter(0);
+      double e0 = f1->GetParError(0);
+      double p1 = f1->GetParameter(1);
+      double e1 = f1->GetParError(1);
+      double p2 = f1->GetParameter(2);
+      double e2 = f1->GetParError(2);
+      double p3 = f1->GetParameter(3);
+      double e3 = f1->GetParError(3);
+      double p4 = f1->GetParameter(4);
+      double e4 = f1->GetParError(4);
+      double p5 = f1->GetParameter(5);
+      double e5 = f1->GetParError(5);
+      double chi2 = f1->GetChisquare();
+      double prob = f1->GetProb();
+
+      edm::LogPrint("LorentzAngle") << std::setprecision(4) << i_module << "\t" << i_layer << "\t" << p0 << "\t" << e0
+                                    << "\t" << p1 << std::setprecision(3) << "\t" << e1 << "\t" << e1 / p1 * 100.
+                                    << "\t" << (p1 - 0.424) / e1 << "\t" << p2 << "\t" << e2 << "\t" << p3 << "\t" << e3
```

Comment:
This corresponds to 23 degree of LA from [simulation.](https://cds.cern.ch/record/1103004/files/NOTE2008_012.pdf )
<img width="731" alt="Screen Shot 2021-10-05 at 12 22 04 AM" src="https://user-images.githubusercontent.com/67766900/135978407-4af43d81-47fe-40d8-8b6c-b401fe31a83f.png">

Reply:
This looks like phase-0.
Does this still apply for Run3 , Phase-1 detector?
In any case please make it a named constant.

Reply:
@wweiphy I asked Morris, 0.424 is close to the one appropriate for a new sensor operated at 150V.
These are the numbers for 2018 MC:
```
BPix map
Layer 1: 0.22344
Layer 2: 0.26942
Layer 3: 0.31882
Layer 4: 0.31882
```
So can you modify the code so that it handles different p1_offset numbers depending on the layer? Also he might give some updated numbers for Run-3 (L1 I expect to be much different)

Reply:
Update:
```
Run-3 Start-Up numbers
Layer 1, Modules 1-8, 0.436848
Layer 2, Modules 1-4, 0.25802
Layer 2, Modules 5-8, 0.25802
Layer 2, New Modules, 0.294044
Layer 3, Modules 1-4, 0.29374
Layer 3, Modules 5-8, 0.31084
Layer 4, Modules 1-4, 0.29944
Layer 4, Modules 5-8, 0.31426
```